zephyrproject-rtos/zephyr
# tests: smp: add some module and integration test cases

Add some module and integration test cases for testing SMP:

There are several test cases added in this PR for testing:

Add module test case:
- test_smp_release_global_lock() and  test_smp_release_global_lock_irq()
   to verify z_smp_release_global_lock() works

Add integration test cases: 
- test_inc_concurrency()
   verify parallelly increase operations will fail if not applying synchronization on SMP.


Ps:  After clarified by Andy in issue #33551, I remove the test_smp_irq_lock_sem() and test_smp_irq_lock_mutex() because it's better to no use them in this way.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>
